cmake/ProcessOptions.cmake

```diff
@@ -541,6 +541,31 @@ function( NEST_PROCESS_WITH_BOOST )
   endif ()
 endfunction()
 
+function( NEST_PROCESS_TARGET_BITS_SPLIT )
+  if ( target-bits-split )
+    if ( ${target-bits-split} STREQUAL "standard" )
+      set( NUM_BITS_RANK 18 )
+      set( NUM_BITS_TID 9 )
+      set( NUM_BITS_SYN_ID 9 )
+      set( NUM_BITS_LCID 27 )
+      set( NUM_BITS_PROCESSED_FLAG 1 )
+    elseif ( ${target-bits-split} STREQUAL "hpc" )
+      set( NUM_BITS_RANK 20 )
+      set( NUM_BITS_TID 10 )
+      set( NUM_BITS_SYN_ID 6 )
+      set( NUM_BITS_LCID 27 )
+      set( NUM_BITS_PROCESSED_FLAG 1 )
+    else()
+      message( FATAL_ERROR "Invalid target-bits-split selected." )
+    endif()
+    set( NUM_BITS_RANK ${NUM_BITS_RANK}U PARENT_SCOPE )
+    set( NUM_BITS_TID ${NUM_BITS_TID}U PARENT_SCOPE )
+    set( NUM_BITS_SYN_ID ${NUM_BITS_SYN_ID}U PARENT_SCOPE )
+    set( NUM_BITS_LCID ${NUM_BITS_LCID}U PARENT_SCOPE )
+    set( NUM_BITS_PROCESSED_FLAG ${NUM_BITS_PROCESSED_FLAG}U PARENT_SCOPE )
+  endif()
+endfunction()
```

Comment:
Exposing this in CMake creates an unnecessary coupling of the NEST code with CMake. For the sake of clarity, the better place for the constants and toggling between them ("standard" and "hpc") is, for example, nest_types.h.